consensus/spos/worker.go

```diff
@@ -306,25 +308,25 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P, _ func(buffToS
 			"prev hash", header.GetPrevHash(),
 		)
 
-		err = wrk.headerSigVerifier.VerifyRandSeed(header)
+		err = header.CheckChainID(wrk.chainID)
 		if err != nil {
-			log.Debug("verify rand seed failed",
+			log.Debug("chain ID mismatch",
 				"error", err)
 			return err
 		}
 
-		err := header.CheckChainID(wrk.chainID)
+		err = wrk.headerSigVerifier.VerifyRandSeed(header)
 		if err != nil {
-			log.Debug("chain ID mismatch",
+			log.Debug("verify rand seed failed",
 				"error", err)
 			return err
 		}
 
-		err = wrk.forkDetector.AddHeader(header, headerHash, process.BHProposed, nil, nil, false)
+		err = wrk.forkDetector.AddHeader(header, headerHash, process.BHProposed, nil, nil)
 		if err != nil {
-			log.Trace("add header in forkdetector", "error", err.Error())
-			//we should not return error here because the other peers connected to self might need this message
-			//to advance the consensus
+			log.Debug("add header to fork detector failed",
+				"error", err)
+			return err
```

Comment:
This was because we didn't check in depth the method AddHeader. Now its ok. Actually if a node returns an error from this interceptor the message would not be spread to the others. So this method should return error only when is clear that the message is malicious. If it is not good for you doesn't mean that is malicious. So, after I checked again method checkBlockBasicValidity called from forkDetector->AddHeader, actually the single point where AddHeader could return with an error, I reached to the conclusion that is really necessary to take into consideration its return, because the checks done there are independently by the node view (wrong genesis time in received header, nonce > round of chronology, and so on)

Reply:
the problem is when we change the AddHeader function to return error if we do not care about the header then this part of the code would return error as well, and the header is not sent forward anymore, which would make it harder for those to sync.We have to be very carefull here. Add comment to AddHeader interface and implementation to only return error when malicious content.

Reply:
Agree with @sasurobert , we have to add comments on AddHeader function documenting when is the case to return an error and what implications it has

Reply:
Actually I would change it back, as even if this is a malicious message on the consensus topic, it would be added in the black list by each node and next time it couldn't affect anything. I think is more safer to not return error here, than to add  a comment in fork detector which could be ignored by a programmer over some time.